source/examples/fibcmd/CMakeLists.txt

```diff
@@ -59,6 +59,9 @@ target_include_directories(${target}
     PRIVATE
     ${DEFAULT_INCLUDE_DIRECTORIES}
     ${PROJECT_BINARY_DIR}/source/include
+    
+    $<TARGET_PROPERTY:${META_PROJECT_NAME}::baselib,INTERFACE_INCLUDE_DIRECTORIES>
```

Comment:
Since our goal was to provide a clean and simple template for cmake, without adding too much additional stuff, we should not begin to bloat things up by adding redundant information like this.

That said, according to the cmake documentation, the INCLUDE_DIRECTORIES target property should exactly include all INTERFACE_INCLUDE_DIRECTORIES of the dependent targets. I also tested this, and it seems to work for me:

Output without the additional include lines:

CHECK -I/workspace/projects/cginternals/cmake-init/build/source/include -I/workspace/projects/cginternals/cmake-init/source/baselib/include -I/workspace/projects/cginternals/cmake-init/build/source/baselib/include -I/workspace/projects/cginternals/cmake-init/source/fiblib/include -I/workspace/projects/cginternals/cmake-init/build/source/fiblib/include --check-config --enable=warning,performance,portability,information,missingInclude --quiet --std=c++11 --verbose --suppress=missingIncludeSystem main.cpp

Output with the additional include lines:

CHECK -I/workspace/projects/cginternals/cmake-init/build/source/include -I/workspace/projects/cginternals/cmake-init/source/baselib/include -I/workspace/projects/cginternals/cmake-init/build/source/baselib/include -I/workspace/projects/cginternals/cmake-init/source/fiblib/include -I/workspace/projects/cginternals/cmake-init/build/source/fiblib/include --check-config --enable=warning,performance,portability,information,missingInclude --quiet --std=c++11 --verbose --suppress=missingIncludeSystem main.cpp

They are exactly the same, so these lines seem to be unnecessary. If your cmake behaves differently, I would consider it a bug (maybe of a specific cmake version?). One possible thing to check: add dependency from the check-XYZ to the XYZ target, because the documentation hints that the target properties may not be fully evaluated before the target is actually executed.